clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -692,6 +692,34 @@ typename scheduler_tmpl_t<RecordType, ReaderType>::scheduler_status_t
 scheduler_tmpl_t<RecordType, ReaderType>::set_initial_schedule(
     std::unordered_map<int, std::vector<int>> &workload2inputs)
 {
+    bool gather_timestamps = false;
+    if (((options_.mapping == MAP_AS_PREVIOUSLY ||
+          options_.mapping == MAP_TO_ANY_OUTPUT) &&
+         options_.deps == DEPENDENCY_TIMESTAMPS) ||
+        (options_.mapping == MAP_TO_RECORDED_OUTPUT &&
+         options_.replay_as_traced_istream == nullptr && inputs_.size() > 1)) {
+        gather_timestamps = true;
+        if (!options_.read_inputs_in_init) {
+            error_string_ = "Timestamp dependencies require read_inputs_in_init";
+            return STATUS_ERROR_INVALID_PARAMETER;
+        }
+    }
+    bool gather_filetype = options_.read_inputs_in_init;
+    // Avoid reading ahead for replay as it makes the input ords not match in tests.
+    if (options_.mapping == MAP_TO_RECORDED_OUTPUT &&
+        options_.replay_as_traced_istream != nullptr)
+        gather_filetype = false;
+    if (gather_filetype || gather_timestamps) {
+        sched_type_t::scheduler_status_t res =
+            get_initial_input_content(gather_timestamps);
```

Comment:
Add comment: why not just do this everytime to make it simpler? What goes wrong if we do that?

Reply:
Added comments on blocking inputs and ordinals.